retrybp/doc_test.go

```diff
@@ -85,3 +86,70 @@ func Example() {
 	// TODO: use result
 	_ = result
 }
+
+// This example demonstrates how to use retrybp with breakerbp.
+func ExampleBreakerErrorFilter() {
+	const timeout = time.Millisecond * 200
+
+	// TODO: use the actual type of your operation.
+	type resultType = int
+
+	// In real code this should be your actual operation you need to retry.
+	// Here we just use a placeholder for demonstration purpose.
+	operation := func() (resultType, error) {
+		return 0, nil
+	}
+
+	breaker := breakerbp.NewFailureRatioBreaker(breakerbp.Config{
+		// TODO: Add breaker configs.
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	const defaultRetries = 2
+	var result resultType
+	err := retrybp.Do(
+		ctx,
+		func() error {
+			// This is the main function call,
+			// wrapped with breaker.Execute to apply circuit breaker on it.
+			res, err := breaker.Execute(func() (interface{}, error) {
+				ret, err := operation()
+				return ret, err
+			})
+			result, _ = res.(resultType)
+			return err
+		},
+
+		// Default retry options, can be overridden by ctx passed in.
+		retry.Attempts(defaultRetries+1), // retries + the first attempt
+		retrybp.CappedExponentialBackoff(retrybp.CappedExponentialBackoffArgs{
+			InitialDelay: time.Millisecond * 10,
+			MaxJitter:    time.Millisecond * 5,
+		}),
+		retrybp.Filters(
+			retrybp.RetryableErrorFilter, // this should always be the first filter.
+			retrybp.BreakerErrorFilter,
+			retrybp.ContextErrorFilter,
+			retrybp.NetworkErrorFilter, // optional: only use this if the requests are idempotent.
+		),
+		retry.OnRetry(func(n uint, err error) {
+			if err != nil {
+				metricsbp.M.Counter("operation.failure").Add(1)
+				log.Errorw(
+					"operation failed",
+					"err", err,
+					"attempt", n,
+				)
+			} else {
+				metricsbp.M.Counter("operation.succeed").Add(1)
+			}
+		}),
+	)
+
+	// TODO: check error
+	_ = err
+	// TODO: use result
+	_ = result
```

Comment:
Instead of saying "TODO", say "In real code, you will need to check your error and use your results"